src/resources/js/main.js

```diff
@@ -0,0 +1,232 @@
+
+            json.results.map((person) => {
+                addPersonToTable(person);
+            });
+        } catch (e) {
+            let p = document.createElement('p');
+            p.innerText = e;
+            section.append(p);
+            hideLoader('dynamic-table');
+        }
+    });
+
+    function createStyles() {
+        let style = document.createElement('style');
+        style.type = 'text/css';
+        style.innerHTML = `
+                        .dynamic-table { margin-left: 30px; }    
+                        .dynamic-content > table, tr, th, td { border: 1px solid black; }
+                        `;
+
+        document.getElementsByTagName('head')[0].appendChild(style);
+    }
+
+    function promiseWait(seconds) {
+        let promise = new Promise((resolve,reject) => {
+            setTimeout(() => { resolve() }, seconds * 1000);
+        });
+        
+        return promise;
+    }
+
+    function addPersonToTable(person) {
+        let table = document.getElementById('dynamic-table');
+        let tr = document.createElement('tr');
+        tr.innerHTML = `
+                        <td>${person.name}</td>
+                        <td>${person.gender}</td>
+                        <td>${person.hair_color}</td>
+                        <td>${person.eye_color}</td>
+                        `;
+        table.append(tr);
+    }
+
+    function showLoader(nodeIdToAppend) {
+        let node = document.getElementById(nodeIdToAppend);
+        
+        let loader = document.createElement('span');
+        loader.id = nodeIdToAppend + '-loader';
+        loader.className = 'loader';
+        node.append(loader);
+    }
+
+    function hideLoader(nodeId) {
+        let loader = document.getElementById(nodeId + '-loader');
+        loader.remove();
+    } 
+
+}
+
+
+function jsDomFunction() {
+    createStyles();
+
+    let main = document.getElementsByTagName('main')[0];
+    
+    let section = document.createElement('section');
+    section.id = 'dynamic-content';
+    section.className = 'dynamic-content';
+    section.innerHTML = `
+                        <ul id="dynamic-list" class='dynamic-list' style="{display: 'block'}">
+                            <li class="dynamic-list-li">Initial LI - No Click</li>
+                        </ul>
+                    `;
+    
+    main.prepend(section);
+
+    let dynamicDiv = document.getElementById('dynamic-content');
+    appendButton(dynamicDiv, 'Add List Item', 'button-add', 'button-add dynamic-list-button', () => { modifyList('dynamic-list', 'dynamic-list-li', 'add'); });
+    appendButton(dynamicDiv, 'Remove List Item', 'button-remove', 'button-remove dynamic-list-button', () => { modifyList('dynamic-list', 'dynamic-list-li', 'remove') });
+    
+    // #region Helper Functions
+    
+    function createStyles() {
+        let style = document.createElement('style');
+        style.type = 'text/css';
+        style.innerHTML = `
```

Comment:
This works, not how I expected, but it's good. I was looking to use `addClassList`/`removeClassList` or also `<element>.style = {}` I think I can be more explicit. I didn't realize you could do what you're doing, which is cool.

Reply:
Understand what you were going for here after our talk and will review the `addClassList` / `removeClassList` APIs as a follow up.